File: OOAD/shopping-cart-app/src/shopping/components/Customer.java
package shopping.components;

import java.util.ArrayList;

public class Customer {
	private static int totalNumberOfid;

	private int id;
	private String name;
	private ArrayList<Order> orders = new ArrayList<Order>();

	public Customer(String name) {
		this.name = name;
		id = generateId();
	}

	public void addCustomer(Order order) {
		orders.add(order);
	}
	
	public ArrayList<Order> getOrderList(){
		return (ArrayList<Order>) orders;
	}

	static {
		totalNumberOfid = 0;
	}

	public int generateId() {
		return (++totalNumberOfid);
	}

	public String getCustomerName() {
		return name;
	}

	public int getCustomerId() {
		return id;
	}
}

File: C#/OOP/ShopingCart-App/ShopingCart-App/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ShopingCart_App
{
    class Customer
    {
        private static int totalNumberOfid;

        private int id;
        private string name;
        private List<Order> orders = new List<Order>();

        static Customer()
        {
            totalNumberOfid = 0;
        }

        public Customer(String name)
        {
            this.name = name;
            id = GenerateId();
        }

        public void AddCustomer(Order order)
        {
            orders.Add(order);
        }

        public List<Order> GetOrderList()
        {
            return (List<Order>)orders;
        }

        public int GenerateId()
        {
            return (++totalNumberOfid);
        }

    }
}
